### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by GrietVermeesch**

### CODE REVIEW

File: autobump-github/src/test/java/com/github/autobump/github/model/GithubReleaseNotesSourceTest.java [hunk 1]
```diff
@@ -84,12 +83,12 @@ void getReleaseNotes_ThrowsGithubUnauthorizedException() {
...
     }
 
-    @Test
+    /*@Test
     void getReleaseNotes_ThrowsGithubNotFoundException() {
         setupErrorStub(404);
         assertThatExceptionOfType(GithubNotFoundException.class)
                 .isThrownBy(() -> githubReleaseNotesSource.getReleaseNotes(TEST_URL, TEST));
-    }
+    }*/
```
Comment (Gschrooyen) [inline autobump-github/src/test/java/com/github/autobump/github/model/GithubReleaseNotesSourceTest.java:91]: this test method should assert that the `githubReleaseNotesSource.getReleaseNotes(TEST_URL, TEST)` method returns `null`

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            mav.addObject("repositoryListDto", new RepositoryListDto(repos));
+        } else {
+            loadRepoOverview(mav);
+        }
+        return mav;
+    }
+
+    @PostMapping("/selectRepositories")
+    public ModelAndView selectRepositories(ModelAndView mav, @ModelAttribute RepositoryListDto dto) {
+        updateSelectedFieldsOfRepos(dto);
```
Comment (Gschrooyen) [inline autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:51]: do this in a service class

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        mav.addObject("repositories", settingsService.getMonitoredRepos());
+        mav.addObject("rep", new RepositoryDto());
+        return mav;
+    }
+
+    @GetMapping("/settings")
+    public ModelAndView settings(ModelAndView mav, @RequestParam("repoId") String repoId) {
+        mav.setViewName("repo-settings");
+        String repoName = settingsService.getRepository(repoId).getName();
+        RepositoryDto dto = settingsService.getSettingsForRepository(repoName);
```
Comment (Gschrooyen) [inline autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:67]: do this in a service class and immediately return the dto from the service method

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        Thread thread = new Thread(() -> autoBumpService.executeAutoBump(repo.getLink()));
+        thread.start();
+    }
+
+    private void updateSelectedFieldsOfRepos(@ModelAttribute RepositoryListDto dto) {
+        for (RepositoryDto repo: dto.getRepositories()
+             ) {
+            settingsService.updateRepo(repo);
+        }
+    }
```
Comment (Gschrooyen) [inline autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:107]: these methods should be in a seperate service class

File: autobump-github/src/test/java/com/github/autobump/github/model/GithubReleaseNotesSourceTest.java (hunk 1 above)
Comment (GrietVermeesch) [reply autobump-github/src/test/java/com/github/autobump/github/model/GithubReleaseNotesSourceTest.java:91]: Done. Have changed the methodname to getReleaseNotes_WhenGithubNotFoundReturnsNull()

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java (hunk 2 above)
Comment (GrietVermeesch) [reply autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:51]: done

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java (hunk 3 above)
Comment (GrietVermeesch) [reply autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:67]: done

File: autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java (hunk 4 above)
Comment (GrietVermeesch) [reply autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:107]: I have moved the creation and execution of the new thread to the AutobumpService class.

Comment (GrietVermeesch) [reply autobump-spring-boot/src/main/java/com/github/autobump/springboot/controllers/SettingsController.java:107]: I have created a new RepoService and renamed the SettingsController to RepoController.

**Pull request merged**